secure_message/common/utilities.py

```diff
@@ -148,6 +148,8 @@ def add_business_details(messages):
 
 def add_users_and_business_details(messages):
     """Add both user and business details to messages based on data from party service"""
+    if messages is None:
+        raise ValueError('messages is required')
```

Comment:
It shouldn't ever be hit if messages=None but if it is then it generates a valueError at exception level with a large stack trace and no context as to what caused it (party_uuid, survey_ref etc), which is what we're trying to move away from.  At this point, it would be a genuine error but could you add some context around it to help diagnose any issues and maybe not dump out the trace.

`{“error”: “ValueError(‘messages is required’,)“, “level”: “exception”, “service”: “ras-secure-message”, “created_at”: “2018-07-19T13:381532003935"}
Traceback (most recent call last):.....`

Reply:
Good catch, @ryangrundy7 worth putting some error logging here, with some key to identify, maybe `url` and `status code` to

Reply:
Good shout Gav 👍

Reply:
The error message could be improved.  If you're looking through the logs and saw 'messages is required'  you'd think that the persons grammar skills are weak.  Once you look at the code you can see why it was written that way, but you want to make log lines reasonably expressive where possible.  Something like 'Cannot add details to a NoneType object'.
As I'm writing this,  I think the line above should be `if not messages:` because what happens if you give this an empty list?  In that case, then the error message could be something like 'There are no messages to process' (don't use that though because I don't quite think that's right).

Reply:
If you give it an empty list, `add_business_details` will make an empty call to the party service (not sure if that will error or just generate useless network traffic... both things are undesirable).  Other ones I think will do the same.  I definitely think you should do `if not messages:` to cover against both None and empty lists being passed to this function

Reply:
I think there is a heck before hand where this is called from, but i guess better safe than sorry

Reply:
also if its an empty list, `if messages` will evaluate to false

Reply:
I think it's a good idea, better to be safe with it